### PR TITLE
Update to C-API

### PR DESCRIPTION
**Pull request by hoytak**
After more review and development on the previous version of the C-API, a number of improvements need to be added: 

- General operator support.
- Better initialization support. 
- Several small bugfixes.

### CODE REVIEW

File: src/capi/default_server_initializer.cpp [hunk 1]
```diff
@@ -1,8 +1,3 @@
-/* Copyright © 2018 Apple Inc. All rights reserved.
```
Comment (znation) [inline src/capi/default_server_initializer.cpp]: I assume this should stay.

File: src/capi/TuriCreate.h [hunk 2]
```diff
@@ -1,8 +1,3 @@
-/* Copyright © 2018 Apple Inc. All rights reserved.
```
Comment (znation) [inline src/capi/TuriCreate.h]: I assume license text should stay.

File: src/capi/TuriCreate.h [hunk 3]
```diff
@@ -103,7 +115,7 @@ void tc_setup_log_location(const char* log_file, tc_error** error);
 
    tc_error *error = NULL;
 
-   tc_create_flexible_type_str("hello", &error);
+   tc_flexible_type* ft = tc_ft_create_from_string("hello", &error);
```
Comment (znation) [inline src/capi/TuriCreate.h]: Will need to call tc_release on ft?

File: src/capi/TuriCreate.h [hunk 4]
```diff
@@ -119,14 +131,18 @@ void tc_setup_log_location(const char* log_file, tc_error** error);
...
- *  The char buffer returned is invalidated by calling tc_error_destroy.
+ *  The char buffer returned is invalidated by calling tc_release.
  */
 const char* tc_error_message(const tc_error* error);
 
 
 /** Destroys any types.
  *
- *  Must be called on all allocated struct types.  
+ *  Must be called on the
```
Comment (znation) [inline src/capi/TuriCreate.h]: Bad docstring merge?

File: test/capi/supervised_learning.cxx [hunk 5]
```diff
@@ -1,5 +0,0 @@
-/* Copyright © 2017 Apple Inc. All rights reserved.
```
Comment (znation) [inline test/capi/supervised_learning.cxx]: I assume we want to keep the copyright text.

File: test/capi/capi_utils.hpp [hunk 6]
```diff
@@ -1,8 +1,3 @@
-/* Copyright © 2017 Apple Inc. All rights reserved.
```
Comment (znation) [inline test/capi/capi_utils.hpp]: I assume we want to keep the copyright text.

File: test/capi/capi_sarray.cxx [hunk 7]
```diff
@@ -1,20 +1,15 @@
-/* Copyright © 2017 Apple Inc. All rights reserved.
```
Comment (znation) [inline test/capi/capi_sarray.cxx]: Keep copyright text.

File: test/capi/capi_parameters.cxx [hunk 8]
```diff
@@ -1,8 +1,3 @@
-/* Copyright © 2017 Apple Inc. All rights reserved.
```
Comment (znation) [inline test/capi/capi_parameters.cxx]: Keep copyright text.

File: test/capi/capi_ndarray.cxx [hunk 9]
```diff
@@ -1,8 +1,3 @@
-/* Copyright © 2017 Apple Inc. All rights reserved.
```
Comment (znation) [inline test/capi/capi_ndarray.cxx]: Keep copyright text.

File: test/capi/capi_models.cxx [hunk 10]
```diff
@@ -1,17 +1,12 @@
-/* Copyright © 2017 Apple Inc. All rights reserved.
```
Comment (znation) [inline test/capi/capi_models.cxx]: Keep copyright text.

File: test/capi/capi_functions.cxx [hunk 11]
```diff
@@ -1,8 +1,3 @@
-/* Copyright © 2017 Apple Inc. All rights reserved.
```
Comment (znation) [inline test/capi/capi_functions.cxx]: Keep copyright text

File: test/capi/capi_flexible_type.cxx [hunk 12]
```diff
@@ -1,8 +1,3 @@
-/* Copyright © 2017 Apple Inc. All rights reserved.
```
Comment (znation) [inline test/capi/capi_flexible_type.cxx]: Keep copyright text

File: test/capi/capi_datetime.cxx [hunk 13]
```diff
@@ -1,8 +1,3 @@
-/* Copyright © 2017 Apple Inc. All rights reserved.
```
Comment (znation) [inline test/capi/capi_datetime.cxx]: Keep copyright text

File: test/capi/capi_classifier_evaluation.cxx [hunk 14]
```diff
@@ -1,15 +1,10 @@
-/* Copyright © 2017 Apple Inc. All rights reserved.
```
Comment (znation) [inline test/capi/capi_classifier_evaluation.cxx]: Keep copyright text

File: test/capi/capi_classifier_evaluation.cxx [hunk 15]
```diff
@@ -1,15 +1,10 @@
...
- * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
- */
 #define BOOST_TEST_MODULE capi_classifier_evaluations
 
 #include <vector>
 #include <boost/test/unit_test.hpp>
 #include <unity/lib/gl_sframe.hpp>
 #include <util/test_macros.hpp>
-#include <capi/TuriCreate.h>
+#include <capi/TuriCore.h>
```
Comment (znation) [inline test/capi/capi_classifier_evaluation.cxx]: This should be TuriCreate.h still?

File: test/capi/basic_capi.cxx [hunk 16]
```diff
@@ -1,8 +1,3 @@
-/* Copyright © 2017 Apple Inc. All rights reserved.
```
Comment (znation) [inline test/capi/basic_capi.cxx]: Keep copyright text

File: test/capi/CMakeLists.txt [hunk 17]
```diff
@@ -1,7 +1,3 @@
-# Copyright © 2017 Apple Inc. All rights reserved.
```
Comment (znation) [inline test/capi/CMakeLists.txt]: Keep copyright text

File: src/capi/TuriCreate.h (hunk 3 above)
Comment (hoytak) [reply src/capi/TuriCreate.h]: Done.

File: src/capi/TuriCreate.h [hunk 18]
```diff
@@ -87,8 +88,25 @@ typedef struct tc_flex_enum_list_struct tc_flex_enum_list;
...
  *
  */
-void tc_setup_log_location(const char* log_file, tc_error** error);
+void tc_init_set_log_location(const char* log_file, tc_error** error);
+
+typedef enum {
+  TURI_LOG_EVERYTHING = 0,
+  TURI_LOG_DEBUG = 1,
+  TURI_LOG_INFO = 2,
+  TURI_LOC_EMPH = 4,
```
Comment (TobyRoseman) [inline src/capi/TuriCreate.h]: What is EMPH?

**Pull request merged**